### PR TITLE
Admin dashboard

### PR DESCRIPTION
**Pull request by imdeeep**
in progress .

### CODE REVIEW

File: frontend/src/Components/Asigntask.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        {
+            name: "Pragya",
+            img: "https://images.pexels.com/photos/3370931/pexels-photo-3370931.jpeg?auto=compress&cs=tinysrgb&w=600&lazy=load",
+            status: "1 task in progress"
+        },
+    ]
+    return (
+        <>
+            <div className={styles.maindiv}>
+                <div className={styles.asigntaskto}>
```
Comment (Abhyudaya-SVVV) [inline frontend/src/Components/Asigntask.tsx]: why are you using the dynamic classes why these are needed using tailwind class

File: frontend/src/Components/Asigntask.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                    <h1>Assign task to</h1>
+                    <FaCaretDown />
+                </div>
+                <div className={styles.membername}>
+                    <div className={styles.search}>
+                        <div><FiSearch /></div>
+                        <input type="text" placeholder='search ' />
+                    </div>
+                    {
+                        membername.map((member)=><div className={styles.member}>
```
Comment (Abhyudaya-SVVV) [inline frontend/src/Components/Asigntask.tsx]: use membername?.map to avoid undefined errors

File: frontend/src/Components/Addtask.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import { FaTag } from "react-icons/fa6";
+import { IoMdPersonAdd } from "react-icons/io";
+import MemberCard from "./MemberCard";
+
+const Addtask = () => {
+    const [subject, setSubject] = useState("");
+    const [details, setDetails] = useState("");
+    const [label, setLabel] = useState("Add Labels"); // Default label
+    const [selectedMember, setSelectedMember] = useState<{
+        name: string;
```
Comment (Abhyudaya-SVVV) [inline frontend/src/Components/Addtask.tsx:13]: define the interface and then use it in the usestate

File: frontend/src/Components/Navbar.tsx [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+          <li
+            className="text-blue-500 font-medium relative"
+            onMouseEnter={() => handleMouseEnter('settings')}
+            onMouseLeave={() => handleMouseLeave('settings')}
+          >
+            <a href="#" className="flex items-center">
+              Settings
+            </a>
+            <div
+              className={`absolute top-full left-0 bg-white shadow-md text-sm ${isHovering.settings ? 'block' : 'hidden'}`}
```
Comment (Vamania69) [inline frontend/src/Components/Navbar.tsx:142]: why you are needed to a hovering state any reasons

File: frontend/src/Components/Projectcard.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+          </div>
+          <div className='flex items-center mt-5 -translate-x-[1.5rem]'>
+            <div className='border border-gray-500 rounded-full translate-x-[25px] z-10'>
+              <img src="https://images.pexels.com/photos/774909/pexels-photo-774909.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1" alt="" className='w-10 h-10 rounded-full object-cover' />
+            </div>
+            <div className='border border-gray-500 rounded-full translate-x-[15px]'>
+              <img src="https://images.pexels.com/photos/415829/pexels-photo-415829.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1" alt="" className='w-10 h-10 rounded-full object-cover' />
+            </div>
+            <div className='border border-gray-500 rounded-full'>
+              <img src="https://images.pexels.com/photos/3760858/pexels-photo-3760858.jpeg?auto=compress&cs=tinysrgb&w=600&lazy=load" alt="" className='w-10 h-10 rounded-full object-cover' />
```
Comment (Vamania69) [inline frontend/src/Components/Projectcard.tsx:28]: use the Image tag only in nextjs we do not prefer the img tag it gives the build error

File: frontend/src/Components/Projectdetail/Details.tsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+
+const Details = () => {
+    return (
+        <>
+            <div className="w-[30%] h-[300px] m-5 shadow-lg rounded-[8px] overflow-y-auto">
+            <h1 className='pt-4 pl-4 text-xl'>Details</h1>
+                <div className='flex items-center gap-3 mx-6 mt-5'>
+                    <img src="https://images.pexels.com/photos/3370931/pexels-photo-3370931.jpeg?auto=compress&cs=tinysrgb&w=600&lazy=load" alt="" className='w-10 h-10 rounded-full object-cover' />
```
Comment (Vamania69) [inline frontend/src/Components/Projectdetail/Details.tsx:9]: Image tag

File: frontend/src/Components/Projectdetail/Member.tsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        {
+            name: "Pragya",
+            img: "https://images.pexels.com/photos/3370931/pexels-photo-3370931.jpeg?auto=compress&cs=tinysrgb&w=600&lazy=load",
+            status: "1 task in progress"
+        },
+    ]
+    return (
+        <>
+            <div className="w-[30%] h-[300px] m-5 shadow-lg rounded-[8px] overflow-y-auto">
+                {membername.map((member) => (
```
Comment (Vamania69) [inline frontend/src/Components/Projectdetail/Member.tsx:19]: use proper naming conventions and memberName?.map

File: frontend/src/Components/Projectdetail/Projectdetailcard.tsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+import ProgressBar from "@ramonak/react-progress-bar";
+
+const Projectdetailcard = () => {
+    return (
+        <>
+            <div className="w-[30%] h-[300px] m-5 shadow-lg rounded-[8px] overflow-y-auto">
+                <h1 className='pt-4 pl-4 text-xl'>Club Management</h1>
+                <div className='p-5'>
+                    <p><small className='text-zinc-600 flex items-center gap-1'><IoMdFlag style={{ opacity: '0.6' }} /> Deadline april 5</small></p>
+                    <p><small className='text-zinc-600'>Track club member website for managing task and improve Productivity.</small></p>
```
Comment (Vamania69) [inline frontend/src/Components/Projectdetail/Projectdetailcard.tsx:12]: you can use the exact text-[#code]

File: frontend/src/pages/index.tsx [hunk 9]
```diff
@@ -1,118 +1,33 @@
...
-          <p className={`m-0 max-w-[30ch] text-sm opacity-50 text-balance`}>
-            Instantly deploy your Next.js site to a shareable URL with Vercel.
-          </p>
-        </a>
-      </div>
-    </main>
-  );
+    <>
+      <Head>
+          <title>Club Management System</title>
```
Comment (Vamania69) [inline frontend/src/pages/index.tsx:15]: title in index?

File: frontend/src/pages/projectdetail.tsx
Comment (Vamania69) [inline frontend/src/pages/projectdetail.tsx:1]: use proper naming of files and each page should be inside of a folder with index

File: frontend/src/Components/Navbar.tsx (hunk 4 above)
Comment (imdeeep) [reply frontend/src/Components/Navbar.tsx:142]: Beacause for the dropdown menu.

File: frontend/src/Components/Projectdetail/Member.tsx (hunk 7 above)
Comment (imdeeep) [reply frontend/src/Components/Projectdetail/Member.tsx:19]: I've used `map` for testing purposes. When the API fetches the data, it goes there for the members. It's just a sample.

File: frontend/src/pages/index.tsx (hunk 9 above)
Comment (imdeeep) [reply frontend/src/pages/index.tsx:15]: ok i will remove this !

File: frontend/src/Components/Projectdetail/Projectdetailcard.tsx (hunk 8 above)
Comment (imdeeep) [reply frontend/src/Components/Projectdetail/Projectdetailcard.tsx:12]: okey got it .

File: frontend/src/Components/Projectcard.tsx (hunk 5 above)
Comment (imdeeep) [reply frontend/src/Components/Projectcard.tsx:28]: okey i will fix that

Comment (imdeeep) [reply frontend/src/Components/Projectcard.tsx:28]: when i am using the `Image` tag the following errors occurred that's why i used `img` tag :- Error: Invalid src prop [(https://images.pexels.com/photos/3370931/pexels-photo-3370931.jpeg?auto=compress&cs=tinysrgb&w=600&lazy=load)](https://images.pexels.com/photos/3370931/pexels-photo-3370931.jpeg?auto=compress&cs=tinysrgb&w=600&lazy=load) on `next/image`, hostname "images.pexels.com" is not configured under images in your `next.config.js`
See more info: https://nextjs.org/docs/messages/next-image-unconfigured-host

Comment (Abhyudaya-SVVV) [reply frontend/src/Components/Projectcard.tsx:28]: form where you are importing the image url

File: frontend/src/pages/index.tsx (hunk 9 above)
Comment (Abhyudaya-SVVV) [reply frontend/src/pages/index.tsx:15]: if done mark it resolved

**Pull request closed without merging**